app/src/main/java/com/example/plannr/students/TableMaker.java

```diff
@@ -0,0 +1,203 @@
+
+    /**
+     * generates ArrayList of all courses to be taken and when to take them
+     * @param year year of first fall session
+     * @return ArrayList of all courses to be taken and when to take them as String
+     */
+    public ArrayList<String> buildTable(int year) {
+        //initialize all needed variables
+        ArrayList<String> taken = new ArrayList<>(StudentUserModel.getInstance().getTakenCourses());
+        ArrayList<String> toBe = new ArrayList<>();
+        ArrayList<CourseHash> fall = new ArrayList<>();
+        ArrayList<CourseHash> winter = new ArrayList<>();
+        ArrayList<CourseHash> summer = new ArrayList<>();
+        ArrayList<String> tooAdd = new ArrayList<>();
+        ArrayList<CourseHash> tooRemove = new ArrayList<>();
+        boolean failSafe = false;
+        int order = 0;
+        int counter = 0;
+
+        //sort courses into when they are available
+        for(CourseHash i: table) {
+            if(i.getCourse().getFallAvailability()) {
+                fall.add(i);
+            }
+            if(i.getCourse().getWinterAvailability()) {
+                winter.add(i);
+            }
+            if(i.getCourse().getSummerAvailability()) {
+                summer.add(i);
+            }
+        }
+
+        while(!table.isEmpty() && !failSafe) {
+            failSafe = true;
+            //all that are availble in fall and can be taken
+            //add those to taken
+            counter = 0;
+
+            //check what fall courses can be taken
+            for(CourseHash i : fall){
+                if(counter >= 6) {
+                    break;
+                }
+                if(canTake(i, taken)) {
+                    tooAdd.add(i.getCourseHash());
+                    tooRemove.add(i);
+                    toBe.add(i.getCourse().getCourseCode() + ":" + order + "-" + "Fall " + year);
+                    failSafe = false;
+                    counter++;
+                }
+            }
+            //remove from all lists courses that were just taken in fall
+            //add to courses that were just taken into taken list
+            taken.addAll(tooAdd);
+            fall.removeAll(tooRemove);
+            winter.removeAll(tooRemove);
+            summer.removeAll(tooRemove);
+            table.removeAll(tooRemove);
+            tooAdd.clear();
+            tooRemove.clear();
+
+            //increment year counter
+            year++;
+
+            counter = 0;
+            order++;
+
+            //repeat for winter courses
+            for(CourseHash i : winter){
+                if(counter >= 6) {
+                    break;
+                }
+                if(canTake(i, taken)) {
+                    tooAdd.add(i.getCourseHash());
+                    tooRemove.add(i);
+                    toBe.add(i.getCourse().getCourseCode() + ":" +  order + "-" + "Winter " + year);
+                    failSafe = false;
+                    counter++;
+                }
+            }
```

Comment:
This could probably be extracted into a function for cleaner code, although if it functionality works then its totally fine for now and this course